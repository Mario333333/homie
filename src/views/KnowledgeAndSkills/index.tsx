import { useContext } from "react";
import Slider from "../../components/Slider";
import app_context from "../../app_context";
import useAxios from "../../hooks/useAxios";
import paint from "../../assets/svg/paint.svg";
import game from "../../assets/svg/game.svg";
import ball from "../../assets/svg/ball.svg";
import music from "../../assets/svg/music.svg";
import movies from "../../assets/svg/movies.svg";
import { SlideData } from "../../utils/ts/componentTypes";
import Context from "../../utils/ts/contextType";

const KnowledgeAndSkills = () => {
  const { data: languages, error: errorLanguages } =
    useAxios<Array<SlideData>>("languages");
  const { data: skills, error: errorSkills } =
    useAxios<Array<SlideData>>("skills");

  const context: Context = useContext(app_context);
  const { labels } = context;

  return (
    <div className="knowledge-skills-container">
      <div className="section">
        <h2 className="title-section">{labels?.languages}</h2>
        {errorLanguages === false &&
          languages !== null &&
          languages.map((item: SlideData, index: number) => {
            return <Slider item={item} key={`languages-${index}`} />;
          })}
      </div>
      <div className="section">
        <h2 className="title-section">{labels?.skills}</h2>
        {errorSkills === false &&
          skills !== null &&
          skills.map((item: SlideData, index: number) => {
            return <Slider item={item} key={`skills-${index}`} />;
          })}
      </div>
      <div className="section interests">
        <h2 className="title-section">{labels?.interests}</h2>
        <div>
          <img data-testid="paint" src={paint} alt="paint" />
          <img data-testid="game" src={game} alt="game" />
          <img data-testid="music" src={music} alt="music" />
          <img data-testid="ball" src={ball} alt="ball" />
          <img data-testid="movies" src={movies} alt="movies" />
        </div>
      </div>
    </div>
  );
};

export default KnowledgeAndSkills;
