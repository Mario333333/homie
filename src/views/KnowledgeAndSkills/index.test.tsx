import KnowledgeAndSkills from "./index";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import { Router, Route, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";
import Context from "../../app_context";

const history = createMemoryHistory();

describe("KnowledgeAndSkills", () => {
  afterEach(cleanup);
  const labels = {
    about_me: "Acerca de mi",
    contact: "Contacto",
    education: "EducaciÃ³n",
    experience: "Experiencia",
    portfolio: "Portafolio",
    knowledge_skills: "Conocimientos y habilidades",
    languages: "Idiomas",
    skills: "Habilidades",
    licenses_certifications: "Licencias y certificaciones",
    interests: "Intereses",
  };

  test("knowledgeAndSkills snapshot", () => {
    const { asFragment } = render(<KnowledgeAndSkills />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("knowledgeAndSkills defined component and content", () => {
    history.push("/knowledgeAndSkills");
    const component = render(
      <Context.Provider value={{ labels }}>
        <Router history={history}>
          <Switch>
            <Route exact path="/knowledgeAndSkills">
              <KnowledgeAndSkills />
            </Route>
          </Switch>
        </Router>
      </Context.Provider>
    );
    expect(component).toBeDefined();
    expect(component.getByText(labels.languages)).toBeDefined();
    expect(component.getByText(labels.skills)).toBeDefined();
    expect(component.getByText(labels.interests)).toBeDefined();
    const paintImage = component.getByTestId("paint");
    expect(paintImage).toHaveAttribute("src", "paint.svg");

    const gameImage = component.getByTestId("game");
    expect(gameImage).toHaveAttribute("src", "game.svg");

    const musicImage = component.getByTestId("music");
    expect(musicImage).toHaveAttribute("src", "music.svg");

    const ballImage = component.getByTestId("ball");
    expect(ballImage).toHaveAttribute("src", "ball.svg");

    const moviesImage = component.getByTestId("movies");
    expect(moviesImage).toHaveAttribute("src", "movies.svg");
  });
});
