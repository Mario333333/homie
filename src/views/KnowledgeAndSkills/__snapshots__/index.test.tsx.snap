// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KnowledgeAndSkills knowledgeAndSkills snapshot 1`] = `
<DocumentFragment>
  <div
    class="knowledge-skills-container"
  >
    <div
      class="section"
    >
      <h2
        class="title-section"
      />
    </div>
    <div
      class="section"
    >
      <h2
        class="title-section"
      />
    </div>
    <div
      class="section interests"
    >
      <h2
        class="title-section"
      />
      <div>
        <img
          alt="paint"
          data-testid="paint"
          src="paint.svg"
        />
        <img
          alt="game"
          data-testid="game"
          src="game.svg"
        />
        <img
          alt="music"
          data-testid="music"
          src="music.svg"
        />
        <img
          alt="ball"
          data-testid="ball"
          src="ball.svg"
        />
        <img
          alt="movies"
          data-testid="movies"
          src="movies.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
